zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -629,6 +640,7 @@ object ZClient {
       headers: Headers,
       body: Body,
       sslConfig: Option[ClientSSLConfig],
+      proxy: Option[Proxy],
```

Comment:
You need to actually use the `proxy` in the implementation of `requestAsync`. Otherwise this isn't doing anything.